federation/pkg/typeadapters/secret.go

```diff
@@ -27,10 +27,10 @@ import (
 	kubeclientset "k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
 )
 
-const secretKind = "secret"
+const SecretKind = "secret"
```

Comment:
better to use camel case for SecretKind, it is the convention used for kube object kind.